src/webgpu/api/validation/encoding/queries/begin_end.spec.ts

```diff
@@ -14,45 +14,91 @@ TODO:
         - }
 `;
 
+import { pbool } from '../../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
 import { ValidationTest } from '../../validation_test.js';
 
 export const g = makeTestGroup(ValidationTest);
 
-g.test('occlusion_query,begin_end')
+g.test('occlusion_query,begin_end_balance')
   .desc(
     `
-Tests that begin/end occlusion query on render pass:
-- begin 0
-- begin 0, end (control case)
-- begin 0, begin 1, end
-- begin 0, end, end
-- end
+Tests that begin/end occlusion queries mismatch on render pass:
+- begin 0, end 1 (begin no queries, then end one query)
+- begin 1, end 0 (begin one query, then end no queries)
+- begin 1, end 1 (begin one query, then end one query)(control case)
+- begin 1, end 2 (begin one query, then end two queries)
+- begin 2, end 1 (begin two queries, then end one query)
```

Comment:
It's okay to omit the "gory details" in this description, so we don't have to keep it perfectly in sync with the case list. Instead can just say "begin n queries, then end m queries, for various n and m".